simapp/state.go

```diff
@@ -12,94 +10,75 @@ import (
-	}
+		case config.GenesisFile != "":
+			appState, simAccs, chainID = AppStateFromGenesisFileFn(r, cdc, config)
 
-	switch {
-	case config.ParamsFile != "" && config.GenesisFile != "":
-		panic("cannot provide both a genesis file and a params file")
+		case config.ParamsFile != "":
+			appParams := make(simulation.AppParams)
+			bz, err := ioutil.ReadFile(config.ParamsFile)
+			if err != nil {
+				panic(err)
+			}
 
-	case config.GenesisFile != "":
-		appState, simAccs, chainID = AppStateFromGenesisFileFn(r, config)
+			cdc.MustUnmarshalJSON(bz, &appParams)
+			appState, simAccs, chainID = AppStateRandomizedFn(simManager, r, cdc, accs, genesisTimestamp, appParams)
 
-	case config.ParamsFile != "":
-		appParams := make(simulation.AppParams)
-		bz, err := ioutil.ReadFile(config.ParamsFile)
-		if err != nil {
-			panic(err)
+		default:
+			appParams := make(simulation.AppParams)
+			appState, simAccs, chainID = AppStateRandomizedFn(simManager, r, cdc, accs, genesisTimestamp, appParams)
 		}
 
-		cdc.MustUnmarshalJSON(bz, &appParams)
-		appState, simAccs, chainID = AppStateRandomizedFn(r, accs, genesisTimestamp, appParams)
-
-	default:
-		appParams := make(simulation.AppParams)
-		appState, simAccs, chainID = AppStateRandomizedFn(r, accs, genesisTimestamp, appParams)
+		return appState, simAccs, chainID, genesisTimestamp
 	}
-
-	return appState, simAccs, chainID, genesisTimestamp
 }
 
 // AppStateRandomizedFn creates calls each module's GenesisState generator function
-// and creates
+// and creates the simulation params
 func AppStateRandomizedFn(
-	r *rand.Rand, accs []simulation.Account, genesisTimestamp time.Time, appParams simulation.AppParams,
+	simManager *module.SimulationManager, r *rand.Rand, cdc *codec.Codec,
+	accs []simulation.Account, genesisTimestamp time.Time, appParams simulation.AppParams,
 ) (json.RawMessage, []simulation.Account, string) {
 
-	cdc := MakeCodec()
 	genesisState := NewDefaultGenesisState()
-
-	var (
-		amount             int64
-		numInitiallyBonded int64
-	)
-
-	appParams.GetOrGenerate(cdc, StakePerAccount, &amount, r,
-		func(r *rand.Rand) { amount = int64(r.Intn(1e12)) })
-	appParams.GetOrGenerate(cdc, InitiallyBondedValidators, &amount, r,
-		func(r *rand.Rand) { numInitiallyBonded = int64(r.Intn(250)) })
-
-	numAccs := int64(len(accs))
-	if numInitiallyBonded > numAccs {
-		numInitiallyBonded = numAccs
+	numInitiallyBonded, amount := RandomizedSimulationParams(appParams, cdc, r, int64(len(accs)))
+
+	input := &module.GeneratorInput{
+		AppParams:    appParams,
+		Cdc:          cdc,
+		R:            r,
+		GenState:     genesisState,
+		Accounts:     accs,
+		InitialStake: amount,
+		NumBonded:    numInitiallyBonded,
+		GenTimestamp: genesisTimestamp,
 	}
 
-	fmt.Printf(
-		`Selected randomly generated parameters for simulated genesis:
-{
-  stake_per_account: "%v",
-  initially_bonded_validators: "%v"
-}
-`, amount, numInitiallyBonded,
-	)
+	simManager.GenerateGenesisStates(input)
```

Comment:
single func to generate all genesis states in the order of creation (on `app.go`). I'll add an additional note on the code too.